spec/lib/link_details_extractor_spec.rb

```diff
@@ -25,6 +25,14 @@
         expect(subject.canonical_url).to eq 'https://foo.com/article'
       end
     end
+
+    context 'when canonical URL is set to "null"' do
+      let(:html) { '<!doctype html><link rel="canonical" href="null" />' }
+
+      it 'ignores the canonical URLs' do
+        expect(subject.canonical_url).to eq 'https://foo.com/article'
```

Comment:
This should read `https://foo.com/article?bar=baz123` since it's the originally-requested URL and nothing says to drop the query string.